pkg/operator/controller.go

```diff
@@ -777,25 +775,44 @@ func (c *Controller) getProviderName(nodeName string) (string, error) {
 	return util.GetProviderName(node.Spec.ProviderID), nil
 }
 
-func (c *Controller) syncMachineConfig(pools []*mcfgv1.MachineConfigPool, labels map[string]string, profile *tunedv1.Profile) error {
+func (c *Controller) syncMachineConfig(labels map[string]string, profile *tunedv1.Profile) error {
 	var (
+		bootcmdline     string
 		kernelArguments []string
 	)
 
+	pools, err := c.pc.getPoolsForMachineConfigLabelsSorted(labels)
+	if err != nil {
+		return err
+	}
+
+	// The following enforces per-pool machineConfigLabels selectors.
+	if len(pools) > 1 {
+		// Log an error and do not requeue, this is a configuration issue.
+		klog.Errorf("profile %v uses machineConfigLabels that match across multiple MCPs (%v); this is not supported",
+			profile.Name, printMachineConfigPoolsNames(pools))
+		return nil
```

Comment:
Should this really return success? What was wrong on returning the error? (the other log update below on line 798)

Reply:
Ack, so the only place we will report this at will be the NTO log?

Reply:
Yes, for now.